File: scripts/yapc2012/hogetarou.rb
#!/usr/bin/env ruby
# coding : utf-8
# Ruby版
# どうもRubyの実行に
primes=[2,3]
primes_sum=5
cnt=2
number=5

while cnt != 10000
  i=1

  while number % (tmp=primes[i]) > 0

  if number < (tmp*tmp)
    primes[cnt] = number
      cnt+=1
      primes_sum+=number
      break
  end
    i+=1
  end

  number+=2
end

puts primes_sum

File: scripts/yapc2012/hogetarou.js
// hogetarou.php hogetarou.rbの
// Javascript版
// 芸がなくてすみません。
// 速度的にはjsが一番早くて、rb,php,perlの順のようです。
// pythonは書けないので未確認
// ただ、言語ごとに得意不得意があると思うので、プログラム次第なんでしょうが
// 計測に使われている環境は rubyは1.9系 PHPは5.3系？のような気がします。

var primes=Array(2,3);
var primes_sum=5;
var cnt=2;

var number=5;
var i;

while (cnt != 10000) {
  i = 1;
  while (number % (tmp=primes[i])) {
    if (number < (tmp*tmp)) {
      primes[cnt++] = number;
      primes_sum+=number;
      break;
    }
    i++;
  }

  number+=2;
}

console.log(primes_sum);
